### PR TITLE
Add bacons consumed score

### PR DESCRIPTION
**Pull request by Saelfen**
#### Summary
Content "Add bacons consumed score"

#### Purpose of change
Add a score to track the number of bacon consumed

#### Describe the solution
Add a score to track the number of bacon consumed. It is important to note that scores are only available on newly made characters, they don't look to proactively load

#### Describe alternatives you've considered
None

#### Testing
Create new character, see bacon count. (yellow because I manually highlighted it)
image

### CODE REVIEW

File: data/json/statistics.json [hunk 1]
```diff
@@ -1756,5 +1756,12 @@
     "stat_type": "count",
     "event_transformation": "avatar_reads_book",
     "description": { "str_sp": "read a book" }
+  },
+  {
+    "id": "num_bacon",
+    "type": "event_statistic",
+    "stat_type": "count",
+    "event_type": "ate_bacon",
+    "description": { "str": "bacon consumed", "str_pl": "bacons consumed" }
```
Comment (Trioct) [inline data/json/statistics.json:1765]: You can avoid the cpp changes by using the already available event transformation avatar_consumes_item
Something like
```json
{
  "id": "avatar_consumes_bacon",
  "type": "event_transformation",
  "event_transformation": "avatar_consumes_item",
  "value_constraints": { "itype": { "equals": [ "itype_id", "bacon" ] } },
  "drop_fields": [ "itype" ]
},
{
  "id": "num_avatar_consumes_bacon",
  "type": "event_statistic",
  "stat_type": "count",
  "event_transformation": "avatar_consumes_bacon",
  "description": { "str": "piece of bacon consumed", "str_pl": "pieces of bacon consumed" }
},
```
instead should simplify things and let this be a json only pr

**Comment by AccountAlias** [comment]
Big thanks to @Trioct for pointing out the use of the existing event transformation

**Comment by actual-nh** [comment]
Might I suggest this for CrazyCataclysm?

**Comment by AccountAlias** [comment]
I'd like this to be part of vanilla if possible instead of integrated into a larger mod. A bacon score doesn't really hurt anything, but that's surely not up for me to decide

**Comment by actual-nh** [comment]
> I'd like this to be part of vanilla if possible instead of integrated into a larger mod. A bacon score doesn't really hurt anything, but that's surely not up for me to decide

I think it may be just me not understanding the craze re bacon.

**Comment by AccountAlias** [comment]
> I think it may be just me not understanding the craze re bacon.

I don't mean to throw names, but I decided to pick it up as requested by Vormithrax since it seemed like a fun and harmless thing to have

**Comment by actual-nh** [comment]
In terms of bacon, in the game one "bacon" (unit) is described as a "thick slab of salty cured bacon", with this making up 5 or 10 strips of bacon (I can never remember re charges). I am not quite getting how achievement score names are generated, but I suggest that it should instead show up as "1 slab of bacon" or "# slabs of bacon". Also, should consuming a "large slab of uncut bacon", from carnivore.json, increase the counter (by 2-3 times)?

**Comment by AccountAlias** [comment]
There are 2 charges per `bacon`, and each charge (a slice) counts as a count against our bacon count. I would certainly argue that a large slab of uncut bacon is not yet bacon, but in fact, cured pork belly 😄 

also oops on the bacons; I updated the description to "pieces of bacon consumed" as per Trioct's suggestion, but never updated the picture in the original post.

**Comment by actual-nh** [comment]
> There are 2 charges per `bacon`, and each charge (a slice) counts as a count against our bacon count.

From the quoted measurements, one charge is then actually 5 slices of bacon.

> I would certainly argue that a large slab of uncut bacon is not yet bacon, but in fact, cured pork belly 😄

Chuckle.

> also oops on the bacons; I updated the description to "pieces of bacon consumed" as per Trioct's suggestion, but never updated the picture in the original post.

Got it.

**Comment by AccountAlias** [comment]
updated to be slabs of bacon instead of slices of bacon

**Comment by Xaleth** [comment]
Yeah I'd go with @actual-nh and say this fits more in Crazy Cataclysm

**Comment by actual-nh** [comment]
> Yeah I'd go with @actual-nh and say this fits more in Crazy Cataclysm

Up to Kevin - he's the one who specifically maintains Crazy Cataclysm (why I put him in as a reviewer).

**Comment by faefux** [comment]
> > I'd like this to be part of vanilla if possible instead of integrated into a larger mod. A bacon score doesn't really hurt anything, but that's surely not up for me to decide
> 
> I think it may be just me not understanding the craze re bacon.

From my understanding it is a long standing reddit meme. I agree that overly 'memey' probably goes in to crazy cataclysm.

**Comment by scorpion451** [comment]
> 
> 
> > > I'd like this to be part of vanilla if possible instead of integrated into a larger mod. A bacon score doesn't really hurt anything, but that's surely not up for me to decide
> > 
> > 
> > I think it may be just me not understanding the craze re bacon.
> 
> From my understanding it is a long standing reddit meme. I agree that overly 'memey' references in the internet sense probably goes in to crazy cataclysm.

It's not a reddit meme, or even really an internet meme, so much as a cultural meme. Other parts of the world have picked up on it via the internet, but in the US and Canada there's a long-standing tradition that bacon is an absolute good and should essentially be treated as a condiment.

**Comment by actual-nh** [comment]
> > From my understanding it is a long standing reddit meme. I agree that overly 'memey' references in the internet sense probably goes in to crazy cataclysm.
> 
> It's not a reddit meme, or even really an internet meme, so much as a cultural meme. Other parts of the world have picked up on it via the internet, but in the US and Canada there's a long-standing tradition that bacon is an absolute good and should essentially be treated as a condiment.

I'm not sure if I would call it a long-standing tradition in the US (no idea about Canada); people putting bacon on everything is no more than 20-30 years AFAIK, depending on where you were.

**Comment by faefux** [comment]
> > > > I'd like this to be part of vanilla if possible instead of integrated into a larger mod. A bacon score doesn't really hurt anything, but that's surely not up for me to decide
> > > 
> > > 
> > > I think it may be just me not understanding the craze re bacon.
> > 
> > 
> > From my understanding it is a long standing reddit meme. I agree that overly 'memey' references in the internet sense probably goes in to crazy cataclysm.
> 
> It's not a reddit meme, or even really an internet meme, so much as a cultural meme. Other parts of the world have picked up on it via the internet, but in the US and Canada there's a long-standing tradition that bacon is an absolute good and should essentially be treated as a condiment.

Source?

**Comment by kevingranade** [comment]
Fine for crazy cataclysm, definitely not by default.

**Pull request closed without merging**